nltk/parse/transitionparser.py

```diff
@@ -41,9 +42,7 @@ def __init__(self, dep_graph):
         """
         # dep_graph.nodes contain list of token for a sentence
         self.stack = [0]  # The root element
-        self.buffer = range(
-            1, len(
-                dep_graph.nodes))  # The rest is in the buffer
+        self.buffer = list(range(1, len(dep_graph.nodes)))  # The rest is in the buffer
```

Comment:
I see you're using `.buffer.pop(0)` in the later code. For Python lists it is O(N) operation which is bad. If you want to add/remove items from both ends then probably collections.deque is more appropriate; if you want to add/remove items only from the beginning it could make sense to use a reversed buffer: popping an item from the end (`buffer.pop()`) is O(1).